src/js/main.js

```diff
@@ -31,4 +32,203 @@ Site = {
+    var _this = this;
+
+    // Make the ajax request
+    $.getJSON(_this.statusUrl, function(data) {
+
+      // Save data in a cache var, just in case
+      _this.streamData = data.status;
+
+      // Check if the stream is offline
+      if(data.status == 'offline') {
+
+        // Trigger the event
+        _this.triggerEvent('streamoffline', data);
+
+      } else if(data.status === 'online')  {  // else  (stream is online)
+
+        // Trigger the event
+        _this.triggerEvent('streamonline', data);
+      }
+    });
+
+  },
+
+  triggerEvent: function(eventName, detail) {
+    var _this =  this;
+
+    console.log('event triggered: ', eventName);
+
+    var customEvent = new CustomEvent(eventName, {
+      detail: detail,
+    });
+
+    document.dispatchEvent(customEvent);
+  },
+
+};
+
+Site.Player = {
+  neverPlayed: true,
+  fadeTime:  2000,
+  streamUrl: 'http://streaming.radio.co/s0b5e9c02c/listen',
+  init: function() {
+    var _this = this;
+
+    // Player DOM elements
+    _this.playerContainer =  document.getElementById('player-container');
+    _this.playerElement =  document.getElementById('player');
+    _this.playerSrc =  document.getElementById('player-src');
+    _this.playButton =  document.getElementById('play-btn');
+    _this.pauseButton =  document.getElementById('pause-btn');
+    _this.streamStatusText =  document.getElementById('stream-status');
+    _this.nowPlayingText =  document.getElementById('now-playing');
+
+    // Bind event handlers
+    _this.handleOnlineStream = _this.handleOnlineStream.bind(_this);
+    _this.handleCanPlay = _this.handleCanPlay.bind(_this);
+    _this.handleOfflineStream = _this.handleOfflineStream.bind(_this);
+
+    // Subscribe to stream events
+    document.addEventListener('streamonline', _this.handleOnlineStream);
+    document.addEventListener('streamoffline', _this.handleOfflineStream);
+
+    // Subscribe to clic on player controls
+    _this.playButton.addEventListener('click', _this.play.bind(_this));
+    _this.pauseButton.addEventListener('click', _this.pause.bind(_this));
+
+  },
+
+  handleOnlineStream: function(event) {
+    var _this = this;
+
+    var data = event.detail;
+
+    // Add class `online` to the player container
+    _this.playerContainer.classList.add('online');
+
+    // Check if player src is empty
+    if (_this.playerSrc.src === '' || _this.playerElement.paused) {
+      // Add src
+      _this.playerSrc.src =  _this.streamUrl;
+
+      // Load src
+      _this.playerElement.load();
+    }
+
+    // Update marquee status
+    _this.streamStatusText.innerHTML = 'Ahora: ';
+
+    // Update Now playing
+    _this.nowPlayingText.innerHTML = data.current_track.title;
+
+    // If the player has never started playing
+    if (_this.neverPlayed) {
+      // We subscribe to the `canplay` event from the player
+      _this.playerElement.addEventListener('canplay', _this.handleCanPlay);
+    }
+  },
+
+  handleCanPlay: function(event) {
```

Comment:
I think this could be called something like `firstPlay` `triggerFirstPlay` or something. I find `handleCanPlay` a bit confusing. Also why trigger this with events? Seems like overkill to make and remove the binds